src/Symfony/Component/Config/Definition/Builder/NodeFinder.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Definition\Builder;
+
+/**
+ * This class finds nodes in a node tree.
+ *
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ */
+class NodeFinder
+{
+    /**
+     * Finds a node defined by $nodePath within the given $rootNode.
+     *
+     * @param string $nodePath The path of the node to find. e.g "doctrine.orm.mappings"
+     */
+    public function find(string $nodePath, NodeDefinition $rootNode): NodeDefinition
+    {
+        if (!$rootNode instanceof ArrayNodeDefinition) {
+            if ($nodePath !== $rootNode->getName()) {
+                throw new \RuntimeException(sprintf('Node with name "%s" does not exist in the given root node "%s"!', $nodePath, $rootNode->getName()));
+            }
+
+            return $rootNode;
+        }
+
+        $explodedNodePaths = explode('.', $nodePath);
```

Comment:
It looks like we do not need all segments, but that we can just fetch the first segment and perform all following statements based on string operations.

Reply:
you also need to account for `setPathSeparator()`